scipy/integrate/tests/test_quadrature.py

```diff
@@ -527,193 +527,97 @@ def func(x):
+    res = np.moveaxis(res, -1, axis)
+    return res
+
+
 class TestCumulativeSimpson:
     x0 = np.arange(4)
     y0 = x0**2
-    
-    @pytest.mark.parametrize(
-        ("y_func", "x", "dx", "initial", "int_y_func"),
-        [
-            (lambda y: y, None, 1.0, 0, lambda y: y**2 / 2),
-            (lambda y: y, None, 0.5, 5, lambda y: y**2 / 2 + 5),
-            (lambda y: y, np.linspace(0, 4, 17), 1.0, None, lambda y: y**2 / 2),
-            (lambda y: y**2, None, 0, None, lambda y: 0 * y),
-        ],
-    )
-    def test_simpson_1d(self, y_func, x, dx, initial, int_y_func):
-        input_x = x
-        if x is None:
-            x = dx * np.arange(17)
-        y = y_func(x)
-        int_y = int_y_func(x)
-        if initial is None:
-            int_y = int_y[1:]
-
-        assert_equal(
-            cumulative_simpson(y, x=input_x, dx=dx, initial=initial),
-            int_y,
-        )
-
-    @pytest.mark.parametrize(
-        ("y_func", "x", "initial", "axis", "int_y_func"),
-        [
-            # Unit tests with inputs for x, dx is None
-            (
-                lambda y: y**2,
-                np.array([[-1, 2, 3, 4], [-1, 2, 4, 8], [-1, 2, 4, 8]]),
-                None,
-                -1,
-                lambda y: y**3 / 3 + 1 / 3,
-            ),
-            (
-                lambda y: y**2,
-                np.array([[0, 2, 3], [0, 2, 4], [0, 2, 4]]),
-                None,
-                -1,
-                lambda y: y**3 / 3,
-            ),
-            (
-                lambda y: y**2,
-                np.array([[1, 2, 3, 4], [1, 2, 4, 8], [1, 2, 4, 8]]),
-                1 / 3,
-                -1,
-                lambda y: y**3 / 3,
-            ),
-            (
-                lambda y: y**2,
-                np.array([[1, 2, 3, 4], [1, 2, 4, 8], [1, 2, 4, 8]]),
-                np.array([[1 / 3], [1 / 3], [1 / 3]]),
-                -1,
-                lambda y: y**3 / 3,
-            ),
-            (
-                lambda y: y**2,
-                np.array([[1, 1, 1, 1], [2, 2, 2, 2], [4, 4, 4, 8]]),
-                np.array([[1 / 3, 1 / 3, 1 / 3, 1 / 3]]),
-                0,
-                lambda y: y**3 / 3,
-            ),
-        ],
-    )
-    def test_simpson_multid(self, y_func, x, initial, axis, int_y_func):
-        y = y_func(x)
-        int_y = int_y_func(x)
-        if initial is None:
-            y_shape = [slice(None)] * y.ndim
-            y_shape[axis] = slice(1, None)
-            int_y = int_y[tuple(y_shape)]
-
-        assert_allclose(
-            cumulative_simpson(y, x=x, axis=axis, initial=initial),
-            int_y,
-        )
 
-    def test_simpson_multid_w_1d_x(self):
-        input_x = [0, 2, 3, 4]
-        x = np.array([input_x]*3)
-        y = x**2
-        int_y = x**3 / 3
-        assert_allclose(
-            cumulative_simpson(y, x=input_x, initial=0),
-            int_y,
-        )
+    @pytest.mark.parametrize('use_dx', (False, True))
```

Comment:
```suggestion
    @pytest.mark.parametrize(('use_dx', 'order'), [(True, 2), (True, 3), (False, 2)])
    @pytest.mark.parametrize('use_initial', (False, True))
    def test_1d(self, use_dx, order, use_initial):
        # Test for exact agreement with polynomial of highest
        # possible order (3 if `dx` is constant, 2 otherwise).
        rng = np.random.default_rng(82456839535679456794)
        n = 10

        # Generate random polynomials and ground truth
        # integral of appropriate order
        order = 3 if use_dx else 2
        dx = rng.random()
        x = (np.sort(rng.random(n)) if order == 2
             else np.arange(n)*dx + rng.random())
        i = np.arange(order + 1)[:, np.newaxis]
        c = rng.random(order + 1)[:, np.newaxis]
        y = np.sum(c*x**i, axis=0)
        Y = np.sum(c*x**(i + 1)/(i + 1), axis=0)

        # Integrate with `cumulative_simpson`
        initial = Y[0] if use_initial else None
        kwarg = {'dx': dx} if use_dx else {'x': x}
        res = cumulative_simpson(y, **kwarg, initial=initial)

        # Compare result against reference
        # If order == 3 and dx is constant, every even value should agree exactly
        # If order == 2, every even value should agree exactly
        if order == 3:
            res = res[::2] if use_initial else res[1::2]
            ref = Y[::2] if use_initial else Y[2::2] - Y[0]
        elif order == 2:
            res = res if use_initial else res[1:]
            ref = Y if use_initial else Y[2:] - Y[0]
        assert_allclose(res, ref)
```

If order == 2, all values should match
If order == 3 and dx = constant, then every other value should match

Reply:
I know. I chose to keep it simple.